src/EditorFeatures/VisualBasicTest/Diagnostics/RemoveAsyncModifier/RemoveAsyncModifierTests.vb

```diff
@@ -0,0 +1,306 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.Testing
+Imports VerifyVB = Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.VisualBasicCodeFixVerifier(
+    Of Microsoft.CodeAnalysis.Testing.EmptyDiagnosticAnalyzer,
+    Microsoft.CodeAnalysis.VisualBasic.RemoveAsyncModifier.VisualBasicRemoveAsyncModifierCodeFixProvider)
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.Diagnostics.RemoteAsyncModifier
+    Public Class RemoveAsyncModifierTests
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveAsyncModifier)>
+        Public Async Function Function_Task() As Task
+            Await VerifyVB.VerifyCodeFixAsync(
+"Imports System.Threading.Tasks
+
+Class C
+    Async Function {|BC42356:Goo|}() As Task
+        If System.DateTime.Now.Ticks > 0 Then
+            Return
+        End If
+
+        System.Console.WriteLine(1)
+    End Function
+End Class",
+"Imports System.Threading.Tasks
+
+Class C
+    Function {|BC42356:Goo|}() As Task
+        If System.DateTime.Now.Ticks > 0 Then
+            Return Task.CompletedTask
+        End If
+
+        System.Console.WriteLine(1)
+        Return Task.CompletedTask
+    End Function
+End Class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveAsyncModifier)>
+        Public Async Function Function_Task_Throws() As Task
+            Await VerifyVB.VerifyCodeFixAsync(
+"Imports System.Threading.Tasks
+
+Class C
+    Async Function {|BC42356:Goo|}() As Task
+        System.Console.WriteLine(1)
+
+        Throw New System.ApplicationException()
+    End Function
+End Class",
+"Imports System.Threading.Tasks
+
+Class C
+    Function {|BC42356:Goo|}() As Task
+        System.Console.WriteLine(1)
+
+        Throw New System.ApplicationException()
+    End Function
+End Class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveAsyncModifier)>
+        Public Async Function Function_Task_WithLambda() As Task
+            Await VerifyVB.VerifyCodeFixAsync(
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Async Function {|BC42356:Goo|}() As Task
+        System.Console.WriteLine(1)
+
+        dim f as Func(of Integer) =
+            Function()
+                Return 1
+            End Function
+    End Function
+End Class",
+"Imports System
+Imports System.Threading.Tasks
+
+Class C
+    Function {|BC42356:Goo|}() As Task
+        System.Console.WriteLine(1)
+
+        dim f as Func(of Integer) =
+            Function()
+                Return 1
+            End Function
```

Comment:
do you have test with single-line VB sub or func lambda?

Reply:
Yes, line 278 and 117 respectively